### DISCUSSION TITLE
How can I get some value (that I defined) inside tailwind.config.js file?

### DISCUSSION THREAD

**Discussion by evgeniy-vashchuk** [tailwindlabs/tailwindcss#12999] [Help] [answered]
Can I somehow get the value that I defined in tailwind.config.js to use it again?

For example, I need next:

`colors: {`
     ` inherit: 'inherit',`
      `current: 'currentColor',`
     ` transparent: 'transparent',`
      `'white': '#FFFFFF',`
      `'black': '#0C1A35',`
     ` 'gray': '#D9DBDD',`
      `'gray-light': '#F4F4F5',`
      `'yellow': '#FFE759',`
      `'orange': '#FF7300',`
      `'primary': theme.colors.orange // here I need to insert some color from above and It must be primary color`
`},`

![image](https://github.com/tailwindlabs/tailwindcss/assets/24987866/f76ee46b-4107-42da-9785-c3a8d4858e71)

**Comment by wongjn** [comment, marked as answer]
Have your colors as an object and then reference that object:

```js
const myColors: {
  inherit: 'inherit',
  current: 'currentColor',
  transparent: 'transparent',
  'white': '#FFFFFF',
  'black': '#0C1A35',
   'gray': '#D9DBDD',
  'gray-light': '#F4F4F5',
  'yellow': '#FFE759',
  'orange': '#FF7300',
};

export default {
  theme: {
    colors: {
      ...myColors,
      primary: myColors.orange,
```

**Reply by evgeniy-vashchuk** [reply]
Thanks! Is this the only way to do it?

**Reply by wongjn** [reply]
I would not think this is the only way to do it.

**Reply by evgeniy-vashchuk** [reply]
I accidentally saw how to do this in a training video on YouTube:

```
colors: {
      inherit: 'inherit',
      current: 'currentColor',
      transparent: 'transparent',
      'white': '#FFFFFF',
      'black': '#0C1A35',
      'gray': '#D9DBDD',
      'gray-light': '#F4F4F5',
      'gray-dark': '#7B7D7E',
      'yellow': '#FFE759',
      'orange': '#FF7300',
      'red': '#EB5757',
      'primary': "theme('colors.orange')",
      'secondary': "theme('colors.black')",
      'danger': "theme('colors.red')"
}
```

**Reply by tylerrrkd** [reply]
@evgeniy-vashchuk  thanks friend
it works well, but this kind of value could not  be used in scss file, like

```scss
--color-primary: theme('colors.orange'); // works
--color-success: theme('colors.primary'); // does not work, will be recognized as plain text
```

```tsx
// but it works well in this situation, it is enough
<p
  className="text-primary"
>
  text
</p>
```
